docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+network.
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `name`: string type, one of "lncc", "ssd" or "gmi".
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    image:
+      name: "lncc" # other options include "lncc", "ssd" and "gmi", for local normalised cross correlation,
+      weight: 0.1
+```
+
+The following are the default image losses. Additional arguments should be added at the
+same indent level:
+
+- `lncc`: Calls a local normalized cross-correlation type loss. Requires the following
+  arguments:
+
+  - `kernel_size`: int, optional, default=9. Kernel size or kernel sigma for
+    kernel_type="gaussian".
+  - `kernel_type`: str, optional, default="rectangular". One of "rectangular",
+    "triangular" or "gaussian"
+
+- `ssd`: Calls a sum of squared differences loss. No additional arguments required.
+
+- `gmi`: Calls a global mutual informatin loss. Requires the following arguments:
+  - `num_bins`: int, optional, default=23. Number of bins for intensity.
+  - `sigma_ratio`: float, optional, default=0.5. A hyper parameter for the gaussian
+    function.
+
+#### Label
+
+The label loss calculates dissimilarity between labels.
+
+All pre-implemented losses can be used as multi-scale or single scale losses.
+Multi-scale losses require a kernel Additionally, all losses can be weighted, so the
+following two arguments are global to all provided losses:
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `scales`: list, or None. Optional argument. If you do not pass this argument (or pass
+  the list [0], the value `null` or an empty value pair), the loss is calculated at a
+  single scale. If you past a list of length > 1, a multi-scale loss will be used.
+  WARNING: an empty list ([]) will raise an error.
+- `kernel`: str, "gaussian" or "cauchy", default "gaussian". Optional argument. Defines
+  the kernel to use for multi-scale losses.
+
+EG.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    label:
+      weight: 1.0
+      name: "dice" # options include "dice", "cross-entropy", "mean-squared", "generalised_dice" and "jaccard"
+      scales: [1, 2]
+```
+
+The default losses require the following arguments. Additional arguments should be added
+at the same indent level:
+
+- `dice`: Calls a Dice loss on the labels, requires the following arguments:
+
+  - `binary`: bool, default is false. If true, the tensors are thresholded at 0.5.
```

Comment:
does it have gradient when true? @mathpluscode if not, we should add the warning here

Reply:
i think yes, like relu, it still yields gradients